### PR TITLE
[Android] Fix: Alarm rings NOW in DateTime.now (in a difference of 1 sec)

### PR DESCRIPTION
**Pull request by dtkdt100**
FIx (#66)

Before:

https://github.com/gdelataillade/alarm/assets/63166757/aabe131c-0b7d-4121-9f90-6f8cd0044957

After

https://github.com/gdelataillade/alarm/assets/63166757/53cc2e48-dfc0-4efa-ae4b-88e21a932e5d

### CODE REVIEW

**Comment by gdelataillade** [comment]
Hi @dtkdt100,

Your added code will only run when the alarm is set to sound at the precise moment it is activated. Is this what you want ? What's the use case ?

**Comment by dtkdt100** [comment]
@gdelataillade For me the use case is:
I am developing an app that do alarm by GPS and when the user will arrive at the right location I want the alarm to ring now.

Also, in your example app there is a button for "Ring now", so I tought if it is in your example app it must be used for something

**Comment by gdelataillade** [comment]
Hi @dtkdt100,

OK I see. The NOW button in the example is to show the proof-of-concept.

**Comment by gdelataillade** [comment]
Did you make some tests ? Your code works good ?

**Comment by dtkdt100** [comment]
@gdelataillade Yes.

Tests:
1. Rings now (as you can see in the video)
2. Set an alarm for one min later (works find, alarm and notificaiton are working good)

**Comment by gdelataillade** [comment]
Does your GPS app has to ring even if the app is on background ?

**Comment by dtkdt100** [comment]
Indeed.
I am using `fullScreenIntent: true` in `FlutterLocalNotificaion` and `AudioSession` in Android or `VolumeController` in iOS in oder to set the volume up as suggested in https://github.com/gdelataillade/alarm/issues/37

**Comment by gdelataillade** [comment]
With the code you added, are you sure the alarm will ring in the background ? Because the point of using `AndroidAlarmManager` is to be able to run the Flutter code even if the app is on background.

**Comment by dtkdt100** [comment]
Yes it is working for me. But i am not sure if it is because of `fullScreenIntent: true`.

But in this case there is not option that the phone will be in background because when you start the alarm the app will be in foreground, the alarm rings immediately

**Comment by gdelataillade** [comment]
OK cool, then give me a few days to run some tests and take a look at `fullScreenIntent: true`. I'll keep you updated. In the meantime you can use your fork for your project.

Thanks for your work.

**Comment by dtkdt100** [comment]
Thanks for your quick response!

File: lib/src/android_alarm.dart [hunk 1]
```diff
@@ -67,6 +67,17 @@ class AndroidAlarm {
       throw AlarmException('Isolate error: $e');
     }
 
+    if (dateTime.difference(DateTime.now()).inSeconds <= 1) {
+      await playAlarm(id, {
+        "assetAudioPath": assetAudioPath,
+        "loopAudio": loopAudio,
+        "vibrate": vibrate,
+        "fadeDuration": fadeDuration.toInt(),
+        "enableNotificationOnKill": enableNotificationOnKill,
```
Comment (gdelataillade) [inline lib/src/android_alarm.dart:76]: We don't need to pass the arguments `vibrate` and `enableNotificationOnKill` right ?

**Review by gdelataillade — CHANGES_REQUESTED** [review]

File: lib/src/android_alarm.dart (hunk 1 above)
Comment (gdelataillade) [reply lib/src/android_alarm.dart:76]: Any update ? Otherwise I can do it myself if you want.

Comment (dtkdt100) [reply lib/src/android_alarm.dart:76]: Oops, didn't notice your comment. 
I am on a trip right now, so I can't do it now.
I will arrive in three days and fix it. It is an easy fix you could do it yourself if you want.

Comment (gdelataillade) [reply lib/src/android_alarm.dart:76]: It's your PR so I can't modify it directly, but no worries I'll wait !

Comment (dtkdt100) [reply lib/src/android_alarm.dart:76]: Thanks!

**Comment by dtkdt100** [comment]
@gdelataillade 
Ready to merge

**Review by gdelataillade — APPROVED** [review]
OK

**Comment by gdelataillade** [comment]
Merged to main branch ! Thanks for your contribution !

**Pull request merged**